packages/qgrid-ngx/src/lib/column/column.component.ts

```diff
@@ -180,8 +182,12 @@ export class ColumnComponent implements OnInit, OnDestroy, OnChanges {
 
   ngOnDestroy() {
     const { column } = this.selfHost;
-    if (column && column.source === 'template') {
-      this.columnList.delete(column.key);
+    if (!(column && column.source === 'template')) {
```

Comment:
`!column || column.source !== 'template'` is more readable, as for me.
If there's no column OR coumn source is not template.
Same thing with column.key.

Btw, why you even need to change this condition?
```typescript
if (column && column.key && column.source === 'template') {
  this.columnList.delete(column.key);
}
````